### DISCUSSION TITLE
How to add menu to content item page

### DISCUSSION THREAD

**Discussion by syedshadab19** [OrchardCMS/OrchardCore#11970] [Q&A] [answered]
Hi I am software as a service and created page using content item I want to add menu to that page so I created a menu for it still it does not show the menu item. As you can see in below snapshot I have added the menu item still its not getting displayed.

![image](https://user-images.githubusercontent.com/25971751/177926482-9195d9f4-b290-494c-b7cd-c4c64193b237.png)

**Comment by hishamco** [comment]
Did you publish both the menu and menu items?

**Reply by syedshadab19** [reply]
yes @hishamco  I have published both still it does not work

**Reply by hishamco** [reply]
Could you please attach a GIF screenshot to check the steps to reproduce

**Reply by syedshadab19** [reply]
Hi @hishamco  I was able to add the menu but only the menu title is getting displayed on page and when clicked on that title it displaying menu items in a new page can you please let me know how to display the menu items in same page please find below give showing reproduction steps

![reproduction_steps](https://user-images.githubusercontent.com/25971751/178674741-4b9107db-f39d-48bc-a956-3b34e319d77d.gif)
.

**Reply by hishamco** [reply]
I though your are talking about admin menu items, but seems you are talking about content menu items. Please refer to @Skrypt comment [below](https://github.com/OrchardCMS/OrchardCore/discussions/11970#discussioncomment-3129589)

**Reply by syedshadab19** [reply]
ok

**Comment by Skrypt** [comment, marked as answer]
This is not how it is supposed to be used. The "Link Menu Item" is not supposed to be added directly to a Content-Type "page".
You can look at the documentation on how to create and use a Menu and integrate it inside a page "content type" template.

**Reply by syedshadab19** [reply]
@Skrypt ok I tried by creating menu, still after creating the menu I am unable to add it to the content item page

**Reply by Skrypt** [reply]
You need to have a Content Type template/view and then you add the menu shape where you need it. The menu data is loaded by "alias". Take a look at our OC Themes and documentation about menus.

**Reply by syedshadab19** [reply]
ok thank you will check it

**Reply by syedshadab19** [reply]
Hi @Skrypt I was able to add the menu but only the menu title is getting displayed on page and when clicked on that title it displaying menu items in a new page can you please let me know how to display the menu items in same page please find below give showing reproduction steps

![reproduction_steps](https://user-images.githubusercontent.com/25971751/178674741-4b9107db-f39d-48bc-a956-3b34e319d77d.gif)
.

**Reply by Skrypt** [reply]
You have a custom "Content Menu Item" ContentPart here which allows you to select a specific menu.
Normally, you don't need this. You just use a menu shape inside the custom Content Type template.
But if that's what you require then you are probably missing the ContentMenuItem ContentPart shape template that will render your menu appropriately.

**Reply by Skrypt** [reply]
https://user-images.githubusercontent.com/3228637/178896300-a61dff86-345a-4e3a-a7d7-fc87e78e62ed.mp4
